### PR TITLE
WIFI 1227 merge endpoints forCustomerWithFilter and searchByMacAndName

### PR DESCRIPTION
**Pull request by ralphlee3**

### CODE REVIEW

File: equipment-service/src/main/resources/equipment-service-openapi.yaml [hunk 1]
```diff
@@ -899,66 +899,21 @@ paths:
       - name: equipmentType
         in: query
         description: equipment type
-        required: true
+        required: false
```
Comment (dmitry-toptygin-connectus-ai) [inline equipment-service/src/main/resources/equipment-service-openapi.yaml:902]: This uncovered an older bug in RDBMS implementation (which was not triggered before because equipmentType was a required parameter) - now the DAO throws NPE if equipmentType is not provided. Please fix that bug in the DAO (I know, you did not create it in the first place).

File: equipment-datastore-rdbms/src/main/java/com/telecominfraproject/wlan/equipment/datastore/rdbms/EquipmentDAO.java [hunk 2]
```diff
@@ -532,21 +532,21 @@ public Equipment delete(long equipmentId) {
...
+                    equipmentType, customerId, locationIds, criteria, context.getLastReturnedPageNumber());
             return ret;
         }
 
-        LOG.debug("Looking up equipment {} for customer {} locations {} last returned page number {}", equipmentType,
-                customerId, locationIds, context.getLastReturnedPageNumber());
+        LOG.debug("Looking up equipment {} for customer {} locations {} criteria {} last returned page number {}", equipmentType,
+                customerId, locationIds, criteria, context.getLastReturnedPageNumber());
 
         String query = SQL_GET_BY_CUSTOMER_ID_AND_EQUIPMENT_TYPE;
```
Comment (dmitry-toptygin-connectus-ai) [inline equipment-datastore-rdbms/src/main/java/com/telecominfraproject/wlan/equipment/datastore/rdbms/EquipmentDAO.java:552]: queryArgs.add(equipmentType.getId()); <- this will throw NPE when equipmentType is null.

To fix this, a query should be constructed differently - when equipmentType is null, it should not be part of the query at all.

Comment (dmitry-toptygin-connectus-ai) [reply equipment-datastore-rdbms/src/main/java/com/telecominfraproject/wlan/equipment/datastore/rdbms/EquipmentDAO.java:552]: also, please add a unit test for equipmentType==null

**Review by dmitry-toptygin-connectus-ai — APPROVED** [review]
